pkg/types/gcp/validation/platform.go

```diff
@@ -53,6 +55,15 @@ func ValidatePlatform(p *gcp.Platform, fldPath *field.Path) field.ErrorList {
 	if _, ok := Regions[p.Region]; !ok {
 		allErrs = append(allErrs, field.NotSupported(fldPath.Child("region"), p.Region, validRegionValues))
 	}
+	if p.PublicZoneID != "" {
+		// PublicZoneID must be 1-63 characters long, must begin with a
+		// letter, end with a letter or digit, and only contain lowercase
+		// letters, digits or dashes.
+		match := publicZoneRegex.MatchString(p.PublicZoneID)
+		if match == false {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("publicZoneID"), p.PublicZoneID, "public zone id is not properly formatted"))
```

Comment:
nit, can we include the expected regex string in the message to allow the user to act upon the error?

Reply:
Fixed.